src/app/visa/c.spec.ts

```diff
@@ -0,0 +1,630 @@
+// write test cases for visa C
+import { calculatePoints, errorMessages } from '@app/domain'
+
+import {
+    academicBackgroundWith,
+    annualSalaryOf,
+    contractingOrganizationOf,
+    japanese,
+    professionalCareerWith,
+    simulationWithCriteriaC,
+    specialOf,
+    universityOf,
+    positionInCompany
+} from '@app/spec.helper'
+
+// non university holder business owner
+// investment banker
+describe('Visa type C point simulation', () => {
+
+    describe('categories', () => {
+        describe('academic background', () => {
+            it('holder of multiple degrees in many areas', () => {
+                const checklist = simulationWithCriteriaC([
+                    academicBackgroundWith({ degree: 'dual_degree' }),
+                    academicBackgroundWith({ degree: 'master' }),
+                ])
+
+                const { matches, points } = calculatePoints(checklist)
+
+                expect(points).toBe(25)
+                expect(matches.map(m => m.id).sort()).toEqual(['dual_degree', 'master'])
+            })
+
+            it('multiple degrees, should pick highest degree', () => {
+                const checklist = simulationWithCriteriaC([
+                    academicBackgroundWith({ degree: 'master' }),
+                    academicBackgroundWith({ degree: 'bachelor' }),
+                ])
+
+                const { matches, points } = calculatePoints(checklist)
+
+                expect(points).toBe(20)
+                expect(matches.map(m => m.id).sort()).toEqual(['master'])
+            })
+
+            it('a dual degree should give bonus', () => {
+                const checklist = simulationWithCriteriaC([
+                    academicBackgroundWith({ degree: 'master' }),
+                    academicBackgroundWith({ degree: 'bachelor' }),
+                    academicBackgroundWith({ degree: 'dual_degree' }),
+                ])
+
+                const { matches, points } = calculatePoints(checklist)
+
+                expect(points).toBe(25)
+                expect(matches.map(m => m.id).sort()).toEqual(['dual_degree', 'master'])
+            })
+        })
+
+        describe('professional career', () => {
+
+            it('10 years of experience', () => {
+                const checklist = simulationWithCriteriaC([
+                    professionalCareerWith({ yearsOfExperience: 10 }),
+                ])
+
+                const { matches, points } = calculatePoints(checklist)
+
+                expect(points).toBe(25)
+                expect(matches.map(m => m.id).sort()).toEqual(['10_years_or_more'])
+            })
+
+            it('8_years_or_more', () => {
```

Comment:
I think it'd be better to test the edge cases here (and also in visa b, bad example from me).

Reply:
done